kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/FilterFactory.java

```diff
@@ -5,20 +5,38 @@
  */
 package io.kroxylicious.proxy.filter;
 
+import java.util.Objects;
+
 import edu.umd.cs.findbugs.annotations.NonNull;
 
 /**
  * FilterFactory is a pluggable source of {@link Filter} instances.
  * @param <F> the {@code Filter} type.
  * @param <C> the type of configuration used to create the {@code Filter}. Use {@link Void} if the {@code Filter} is not configurable.
  */
-public interface FilterFactory<F extends Filter, C> {
+public abstract class FilterFactory<F extends Filter, C> {
+
+    private final Class<C> configType;
+    private final Class<F> filterType;
+
+    /**
+     * Constructor
+     * @param configType The concrete class of {@code Filter} this factory {@linkplain #createFilter(FilterCreationContext, C) creates}.
+     * @param filterType The type of configuration this factory requires.
+     * Use {@code Void.class} if the factory does not support configuration.
+     */
+    public FilterFactory(Class<C> configType, Class<F> filterType) {
```

Comment:
I'd reverse the order of the formal parameters to make the ordering of the parameters of the class itself.

Reply:
Use the @NonNull annotation on the parameters too.